sky/shell/platform/android/org/domokit/sky/shell/SkyApplication.java

```diff
@@ -63,10 +69,29 @@ protected void onBeforeResourceExtraction(ResourceExtractor extractor) {
         extractor.addResources(SKY_RESOURCES);
     }
 
+    void registerService(ServiceRegistry registry, String classPath) {
+        try {
+            Class.forName(classPath)
+                .getMethod("onServiceRegistryAvailable", ServiceRegistry.class)
+                .invoke(null, registry);
+        } catch(Exception e) {
+            Log.e(TAG, "Failed to register service '" + classPath + "'", e);
+        }
+    }
+
     /**
       * Override this function to register more services.
       */
     protected void onServiceRegistryAvailable(ServiceRegistry registry) {
+        final AssetManager manager = getResources().getAssets();
+        try (BufferedReader reader = new BufferedReader(new InputStreamReader(manager.open(SERVICES)))) {
+            String line = null;
+            while ((line = reader.readLine()) != null)
+                registerService(registry, line);
```

Comment:
Might want to filter out blank lines here, and possibly also comments.  If the file gets more complex the parsing might want to be pulled out into a separate method.